rmw_fastrtps_shared_cpp/src/rmw_publisher.cpp

```diff
@@ -50,41 +47,32 @@ __rmw_destroy_publisher(
     return RMW_RET_ERROR;
   }
 
-  if (!publisher) {
-    RMW_SET_ERROR_MSG("publisher handle is null");
-    return RMW_RET_ERROR;
-  }
-
-  if (publisher->implementation_identifier != identifier) {
-    RMW_SET_ERROR_MSG("publisher handle not from this implementation");
-    return RMW_RET_ERROR;
-  }
```

Comment:
Moved https://github.com/ros2/rmw_fastrtps/blob/a542af36cf1cf4b6003a73b59d3363a432b271ce/rmw_fastrtps_shared_cpp/src/publisher.cpp#L38-L49.
Those arguments are passed without being used, so it's fine to check them in the called function.

Reply:
Right, but you dereference `publisher` in [line 51](https://github.com/ros2/rmw_fastrtps/pull/312/files#diff-1623b37ee69464abc9f38e1afc71215bR51) in hopes it's not `nullptr` but a Fast-RTPS publisher so that the cast makes sense.

Reply:
This still applies.

Reply:
See https://github.com/ros2/rmw_fastrtps/pull/312/commits/9fd7fe605ca749201371a4636274c2bfdb8ec1d0.